File: server/src/models/Verse.js
const mongoose = require('mongoose');

const verseSchema = new mongoose.Schema({
  chapterId: {
    type: String,
    required: true,
    index: true,
  },
  verseNumber: {
    type: Number,
    required: true,
  },
  textSanskrit: {
    type: String,
    required: true,
  },
  textTransliteration: {
    type: String,
    required: true,
  },
  translationEnglish: {
    type: String,
    required: true,
  },
  translationHindi: {
    type: String,
    required: true,
  },
  explanation: {
    type: String,
    required: true,
  },
  videoUrl: {
    type: String,
    default: '',
  },
  audioUrl: {
    type: String,
    default: '',
  },
  images: [{
    type: String,
  }],
}, {
  timestamps: true,
  toJSON: { 
    virtuals: true,
    transform: function(doc, ret) {
      ret.id = `${ret.chapterId}-${ret.verseNumber}`;
      return ret;
    }
  },
  toObject: { virtuals: true },
});

// Compound index for efficient queries
verseSchema.index({ chapterId: 1, verseNumber: 1 }, { unique: true });

module.exports = mongoose.model('Verse', verseSchema);

File: backup/geeta book app copy 2/lib/shared/domain/models/verse.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'verse.freezed.dart';
part 'verse.g.dart';

@freezed
class Verse with _$Verse {
  const factory Verse({
    required String id,
    required String chapterId,
    required int verseNumber,
    required String textSanskrit,
    required String textTransliteration,
    required String translationEnglish,
    required String translationHindi,
    required String explanation,
    String? videoUrl,
    String? audioUrl,
    @Default([]) List<String> images,
    @Default(false) bool isCompleted,
    DateTime? completedAt,
  }) = _Verse;

  factory Verse.fromJson(Map<String, dynamic> json) =>
      _$VerseFromJson(json);
}
